### PR TITLE
[9.4.x] REPL read opt perf

### PR DESCRIPTION
**Pull request by wburns**

### CODE REVIEW

**Comment by wburns** [comment]
run performance tests please
library repl reads

**Comment by diegolovison** [comment]
run performance tests please
library repl reads

**Comment by diegolovison** [comment]
run performance tests please
library repl reads

**Comment by diegolovison** [comment]
run performance tests please
library repl reads

**Comment by diegolovison** [comment]
run performance tests please
library repl reads

**Comment by wburns** [comment]
reverted to get baseline with 9.4.x as well.

**Comment by wburns** [comment]
run performance tests please
library repl reads

**Comment by diegolovison** [comment]
run performance tests please
library repl reads

**Comment by wburns** [comment]
run performance tests please
library repl reads

**Comment by wburns** [comment]
run performance tests please
library repl reads
cs-hotrod repl reads

**Comment by diegolovison** [comment]
run performance tests please
library repl reads
cs-hotrod repl reads

**Comment by wburns** [comment]
run performance tests please
library repl reads

**Comment by wburns** [comment]
run performance tests please
library repl reads

**Comment by diegolovison** [comment]
@wburns 
http://perfrepo.mw.lab.eng.bos.redhat.com/reports/tableComparisonReport/18129
http://perfrepo.mw.lab.eng.bos.redhat.com/reports/tableComparisonReport/18130

**Comment by wburns** [comment]
Odd that is shows no improvement. I have a JMH test that I have been running that shows almost a 25% increase in throughput for get with the if branch commented out. I will have to look closer in a bit.

**Comment by wburns** [comment]
run performance tests please
library repl reads

**Comment by wburns** [comment]
Performance seems pretty good in the latest run. For some reason mean response time is still higher, however the actual throughput is higher. I will try another to see what it says.

**Comment by wburns** [comment]
run performance tests please
library repl reads

**Comment by wburns** [comment]
Rebased and should be ready if it is deemed okay.

**Comment by wburns** [comment]
run performance tests please
library repl reads

**Comment by tristantarrant** [comment]
Merged

File: core/src/main/java/org/infinispan/factories/InternalCacheFactory.java [hunk 1]
```diff
@@ -311,56 +312,18 @@ public boolean containsKey(Object key) {
...
-               }
-               return CompletableFuture.completedFuture(ice);
-            } else {
-               return stage.thenApply(expired -> {
-                  if (expired == Boolean.TRUE) {
-                     return null;
-                  }
-                  return (CacheEntry<K, V>) ice;
-               }).toCompletableFuture();
+            if (CompletionStages.join(stage)) {
```
Comment (danberindei) [inline core/src/main/java/org/infinispan/factories/InternalCacheFactory.java]: Shouldn't we check `if (isCompletedSuccessfully(stage))` first?

**Comment by wburns** [comment]
No need to. This is a blocking method call and there is no lambda to worry
about. I am guessing the diff probably tricked you into thinking it was a
non blocking method 😁

**Comment by danberindei** [comment]
It did :)

**Pull request merged**